File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/api/ApiTestRequest.java
package com.github.seratch.jslack.api.methods.request.api;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

public class ApiTestRequest implements SlackApiRequest {

    /**
     * example property to return
     */
    private String foo;

    /**
     * Error response to return
     */
    private String error;

    ApiTestRequest(String foo, String error) {
        this.foo = foo;
        this.error = error;
    }

    public static ApiTestRequestBuilder builder() {
        return new ApiTestRequestBuilder();
    }

    @Override
    public String getToken() {
        return null;
    }

    public String getFoo() {
        return this.foo;
    }

    public String getError() {
        return this.error;
    }

    public void setFoo(String foo) {
        this.foo = foo;
    }

    public void setError(String error) {
        this.error = error;
    }

    public static class ApiTestRequestBuilder {
        private String foo;
        private String error;

        ApiTestRequestBuilder() {
        }

        public ApiTestRequest.ApiTestRequestBuilder foo(String foo) {
            this.foo = foo;
            return this;
        }

        public ApiTestRequest.ApiTestRequestBuilder error(String error) {
            this.error = error;
            return this;
        }

        public ApiTestRequest build() {
            return new ApiTestRequest(foo, error);
        }

        public String toString() {
            return "ApiTestRequest.ApiTestRequestBuilder(foo=" + this.foo + ", error=" + this.error + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/api/ApiTestRequest.kt
package work.socialhub.kslack.api.methods.request.api

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class ApiTestRequest(
    /** Authentication token bearing required scopes. */
    override var token: String? = null,
    /** example property to return */
    var foo: String?,
    /** Error response to return */
    var error: String?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            it.addParam("foo", foo)
            it.addParam("error", error)
        }
    }
}